pyfarm/jobtypes/core/internals.py

```diff
@@ -792,6 +797,39 @@ def rmdir(path, on_exit_handler=True):
         for directory in self._tempdirs:
             reactor.callInThread(rmdir, directory)
 
+    def _ensure_free_space_in_temp_dir(self, tempdir, space):
```

Comment:
What do you think of adding an optional `minimum_age` keyword?  The idea being that if provided a timedelta object only files older than `minimum_age` would be cleaned up and all others would be ignored.  I realize this should cleanup based on atime but in some situations, such as when a render runs on an artist's machine, I'm not sure that alone will be sufficient.